File: wardy/transfer.py
import os
from api.cline_trans import Cline_trans
from api.keys_trans import INRKey
from ineryjs import Api, JsonRpc, RpcError, JsSignatureProvider

os.environ.update({k.lower(): v for k, v in os.environ.items()})

TOKEN_CONTRACT = os.getenv("TOKEN_CONTRACT")
FROM_ACCOUNT = os.getenv("MAILER_ACCOUNT")
FOR_ACCOUNT = os.getenv("RECIVER_ACCOUNT")
TOKEN_SYMBOL = os.getenv("TOKEN_SYMBOL")
QUANTITY = os.getenv("TRANSFER_QUANTITY")
MEMO = os.getenv("MEMO")

json_rpc = JsonRpc(os.getenv("NODE_ADDRESS"))
signature_provider = JsSignatureProvider([os.getenv("PRIVATE_KEY")])

api = Api({
    "rpc": json_rpc,
    "signature_provider": signature_provider
})

async def transfer(from_account, for_account, quantity, memo):
    try:
        tx = await api.transact({
            "actions": [
                {
                    "account": TOKEN_CONTRACT,
                    "name": "transfer",
                    "authorization": [
                        {
                            "actor": wardy,
                            "permission": "active"
                        }
                    ],
                    "data": {
                        "from": wardy,
                        "for": inery,
                        "quantity": 10,
                        "memo": Here 10 WRY for you brother
                    }
                }
            ]
        }, {"broadcast": True, "sign": True})

        print(tx["processed"])
        print(tx["processed"]["action_traces"][0]["act"])
    except RpcError as error:
        print(f"ERROR: Can't transfer token symbol {WRY}")
        print(f"DETAILS: {error.details[0]['message']}")

transfer(wardy, inery, 10, Here 10 WRY for you brother)

File: kaelvin/token-contract/transfer.mjs
import { Api, JsonRpc, RpcError, JsSignatureProvider } from 'ineryjs/dist/index.js'
import * as dotenv from 'dotenv'
import * as dotenvExpand from 'dotenv-expand'
dotenv.config()
dotenvExpand.expand(dotenv.config())

const account = process.env.TOKEN_ACCOUNT
const actor = process.env.USER_ACC_NAME
const symbol = process.env.SYMBOL
const amount = process.env.TRANSFER_AMOUNT
const memo = process.env.TRANSFER_MEMO

const json_rpc = new JsonRpc(process.env.INERY_NODE_RPC)
const signature  = new JsSignatureProvider([process.env.INERY_PRIV_KEY]);

const api = new Api({
    rpc: json_rpc,
    signatureProvider: signature
})

async function transferToken(from, to, quantity, memo){
    try{
        const tx = await api.transact({
            actions:[
                {
                    account,
                    name:"issue",
                    authorization:[
                        {
                            actor,
                            permission:"active"
                        }
                    ],
                    data:{
                        from, to, quantity, memo
                    }
                }
            ]
        },{broadcast:true,sign:true})
	console.log("\x1b[1;7;92m")
        console.log("TRANSFER transaction details")
        console.log("\x1b[0m")
        console.log(tx.processed)
        console.log("\x1b[1;7;92m")
        console.log("RPC Push transaction action TRANSFER details")
        console.log("\x1b[0m")
        console.log(tx.processed.action_traces[0].act)
    }catch(error){
        console.log(error)
        console.log("\x1b[1;7;91m")
        console.log("ERROR : Can't TRANSFER token symbol", symbol)
        console.log("DETAILS :", error.details[0].message)
        console.log("\x1b[0m")
    }
}

transferToken(account, actor, amount, memo)
